stdlib/shelve.pyi

```diff
@@ -1,35 +1,39 @@
-import collections.abc
 from _typeshed import Self
-from typing import Any, Dict, Iterator, Optional, Tuple
+from collections.abc import Iterator, MutableMapping
+from types import TracebackType
+from typing import Optional, Type, TypeVar, Union, overload
 
-class Shelf(collections.abc.MutableMapping[Any, Any]):
+_T = TypeVar("_T")
+_VT = TypeVar("_VT")
+
+class Shelf(MutableMapping[str, _VT]):
     def __init__(
-        self, dict: Dict[bytes, Any], protocol: Optional[int] = ..., writeback: bool = ..., keyencoding: str = ...
+        self, dict: MutableMapping[str, _VT], protocol: Optional[int] = ..., writeback: bool = ..., keyencoding: str = ...
```

Comment:
The keys and values are both bytes. It always encodes keys before putting them in the underlying dict. The values are pickled representations of the logical values (you can see that in the `__getitem__` implementation).
```suggestion
        self, dict: MutableMapping[bytes, bytes], protocol: Optional[int] = ..., writeback: bool = ..., keyencoding: str = ...
```

Reply:
My bad. I did not read the implementation carefully enough. The fix is done.